router/env/ctrl.go

```diff
@@ -173,3 +189,14 @@ func (self *networkControllers) Close() error {
 	})
 	return errors.ToError()
 }
+
+func (self *networkControllers) CloseAndRemoveByAddress(address string) error {
+	ctrlId, ch := self.GetCtrlChannelByAddress(address)
+	if ch != nil {
+		if err := ch.Close(); err != nil {
+			return err
```

Comment:
I think probably always want to delete, even if there is an error

Reply:
Can do!